client/syntaxes/bitbake.tmLanguage.json

```diff
@@ -1,270 +1,203 @@
+                },
+                "3": {
+                    "name": "comment.line.number-sign.bb"
+                },
+                "4": {
+                    "name": "comment.line.text.bb"
+                }
             }
-          },
-          "match": "(\\$)([a-zA-Z0-9_:])*\\b",
-          "name": "variable.language.bb"
         },
-        {
-          "captures": {
-            "1": {
-              "name": "punctuation.definition.variable.bb"
+        "functions": {
+            "begin": "\\s*(\\s+[a-zA-Z0-9_-][\\w-]*)?(\\.)?([a-zA-Z0-9_-][\\w-]*)(:append|:prepend)?(?=\\s*\\()",
+            "beginCaptures": {
+                "1": {
+                    "name": "variable.other.names.bb"
+                },
+                "2": {
+                    "name": "keyword.operator.bb"
+                },
+                "3": {
+                    "name": "entity.name.function.python.bb"
+                },
+                "4": {
+                    "name": "keyword.other.bitbake-operator.bb"
+                }
+            },
+            "end": "(?=\\))",
+            "patterns": [
+                {
+                    "include": "#function-params"
+                }  
+            ]
+        },
+        "function-content": {
+            "begin": "(?<=\\(.*\\)\\{)",
+            "end": "(?=\\})",
+            "patterns": [
+                {
+                    "include": "#variable-assignment"
+                },
+                {
+                    "include": "#string"
+                },
+                {
+                    "include": "#variable-reference"
+                }
+            ]
+        },
+        "variable-reference": {
+            "patterns": [
+                {
+                    "match": "(\\$\\{)([a-zA-Z0-9_]*)(\\})",
+                    "name": "variable.language.bb",
+                    "captures": {
+                        "1": {
+                            "name": "variable.language.bb"
+                        },
+                        "2":{
+                            "name": "variable.other.bb"
+                        }
+                    }
+                }
+            ]
+        },
+        "variable-name": {
+            "match": "\\s*([a-zA-Z0-9_-][\\w-]*)(:append|:prepend:remove)?(?!\\s*\\()",
+            "captures": {
+                "1": {
+                    "name": "variable.other.names.bb"
+                },
+                "2": {
+                    "name": "keyword.other.bitbake-operator.bb"
+                }
             }
-          },
-          "match": "(\\$\\{)(?:[a-zA-Z0-9_])*(\\})",
-          "name": "variable.language.bb"
+        },
+        "function-params":{
+            "begin": "(?<=\\()",
+            "end": "(?=\\))",
+            "patterns": [
+                {
+                    "include": "#variable-name"
+                },
+                {
+                    "include": "#variable-assignment"
```

Comment:
There is no variable-assignment object defined anymore. Therefore this line has no effect and can be removed.